File: Client/WebApp/DTOs/Company/UpdateCompanyDto.cs
using System.ComponentModel.DataAnnotations;

namespace WebApp.DTOs.Company;

public class UpdateCompanyDto
{
    public long Id { get; set; }
    [Required(ErrorMessage = "Company name is required")]
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Website { get; set; }

    [Required(ErrorMessage = "City is required")]
    public string City { get; set; } = string.Empty;
    public string? Postcode { get; set; }
    public string? Address { get; set; }

    public long CompanyRepresentativeId { get; set; }
}

File: DataAccessLayer/DatabaseServer/src/main/java/com/example/databaseserver/DTOs/UpdateCompanyDto.java
package com.example.databaseserver.DTOs;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateCompanyDto {
    private long id;
    private String name;
    private String description;
    private String website;
    private String city;
    private String postcode;
    private String address;
    private long companyRepresentativeId;
}
